src/Microsoft.Extensions.Configuration.AzureAppConfiguration/StartupOptions.cs

```diff
@@ -0,0 +1,24 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+//
+
+using System;
+
+namespace Microsoft.Extensions.Configuration.AzureAppConfiguration
+{
+    /// <summary>
+    /// Options used when initially loading data into the configuration provider.
+    /// </summary>
+    public class StartupOptions
+    {
+        /// <summary>
+        /// The maximum delay before timing out when loading data from Azure App Configuration on startup.
+        /// </summary>
+        public TimeSpan Timeout { get; set; } = TimeSpan.FromMinutes(1);
```

Comment:
Default timeout of one minute seems a bit low. I'd go with 10 minutes.

Reply:
I'm a bit concerned that we may get complaints like [https://github.com/Azure/AppConfiguration/issues/255](https://github.com/Azure/AppConfiguration/issues/255) all over again if we set the default startup timeout to anything more than a minute.